network/benches/socket_bench.rs

```diff
@@ -306,5 +307,80 @@ fn socket_bench(c: &mut Criterion) {
     c.bench("socket_send_throughput", bench);
 }
 
-criterion_group!(network_benches, socket_bench);
+/// Concurrently dial into server
+fn bench_client_connection<F, T, S>(
+    b: &mut Bencher,
+    concurrency: u64,
+    transport_func: F,
+    server_addr: NetworkAddress,
+) where
+    F: Fn() -> T,
+    T: Transport<Output = S> + 'static,
+    S: AsyncRead + AsyncWrite + Unpin + Send + 'static,
+{
+    let mut runtime = Builder::new()
+        .threaded_scheduler()
+        .core_threads(32)
+        .enable_all()
+        .build()
+        .unwrap();
+    b.iter_with_setup(
+        || {
+            stream::iter(0..concurrency).for_each_concurrent(None, |_| async {
+                let transport = transport_func();
+                // try to establish one successful connection
+                while let Ok(fut) = transport.dial(server_addr.clone()) {
+                    match fut.await {
```

Comment:
same thing here, with noise the dial future is (relatively) CPU intense, so we probably want to just spawn them all

```rust
stream::iter(0..concurrency).for_each_concurrent(None, |_| {
    // this will return a `JoinHandle` future, so the outer for_each will only finish
    // when all tasks have completed
    executor.spawn(async move {
        transport.dial(server_addr.clone()).unwrap().await.unwrap();
    })
})
```

could also try with `BoundedExecutor` pattern, not sure which is faster : p